src/lib/mathlib/math/filter/AlphaFilter.hpp

```diff
@@ -73,19 +73,18 @@ class AlphaFilter
 		}
 	}
 
-	void setCutoffFreq(float sample_freq, float cutoff_freq)
+	bool setCutoffFreq(float sample_freq, float cutoff_freq)
 	{
 		if ((sample_freq <= 0.f) || (cutoff_freq <= 0.f) || (cutoff_freq >= sample_freq / 2.f)
 		    || !isFinite(sample_freq) || !isFinite(cutoff_freq)) {
 
-			// No filtering
-			_alpha = 1.f;
-			_cutoff_freq = 0.f;
-			return;
+			// Invalid parameters
```

Comment:
I'm not sure I have an opinion, but just curious why not have the filter disable itself here rather than handling it externally?

Reply:
The filter should not disable itself when you try to configure it because it's a silent failure. I was assuming the external handling for the IMU filter does disable it because if this happens in air you rather have no filter than no data anymore. But we could also preserve the parameters instead of diabling the filter (would maybe be safer in air) and if I misinterpreted and the cutoff doesn't change in air I would zero out the data instead of silently disable the filter.